### PR TITLE
[RDY] Add focused camera

### PR DESCRIPTION
**Pull request by tinukedaya**
I wanted to create an alternative to security camera by adding another camera with buffed range, but limited field of view. The problem is that I can't really code. I was able to setup the camera in the .json files and add a recipe for it to a book, but to limit the field of vision (to a cone similar to headlights) is out of my reach.

Testing this camera on my copy here works as I intended (except for the field of vision), making fully closed APC's actually viable, while having windows on the vehicle is still better. 

Need help figuring the cone. If I can manage that at all...

### CODE REVIEW

File: data/json/items/vehicle_parts.json [hunk 1]
```diff
@@ -895,6 +895,20 @@
     },
     {
         "type":"GENERIC",
+        "id" : "hdcamera",
+        "name" : "focused camera",
+        "description" : "A security camera with high quality lense assembly you could connect to a display. Allows for greater distance than regular security camera, but the field of vision is limited.",
```
Comment (Night-Pryanik) [inline data/json/items/vehicle_parts.json]: "lense" -> "lenses". 
Also, double space after the period.

**Comment by OzoneH3** [comment]
This seems rather hard to implement since you would have to set the direction the camera is facing and save that somewhere.

The calculation is done in lightmap.cpp line 792. There is this comment inside:
```
// @todo: Factor in the mirror facing and only cast in the
// directions the player's line of sight reflects to.
//
// The naive solution of making the mirrors act like a second player
// at an offset appears to give reasonable results though.
```

The comment seems to suggest, that current cameras and mirrors give a 360° view. So all of that would have to be written as well for custom FOV cameras to work.

**Comment by tinukedaya** [comment]
Oooh, that seems way out of my league! 
In that case i would just call this done.

File: data/json/recipes/recipe_electronics.json [hunk 2]
```diff
@@ -2215,6 +2215,39 @@
...
+  "time": 90000,
+  "book_learn": [ ["textbook_robots", 6] ],
+  "autolearn": false,
+  "qualities":[
+    {"id":"SCREW_FINE","level":1},
+    {"id":"SAW_M","level":1}
+  ],
+  "tools": [
+    [
+      [ "toolset", 10 ]
```
Comment (Cyrano7) [inline data/json/recipes/recipe_electronics.json]: This is the actually the bionic toolset so you might want to add the soldering iron also.

File: data/json/items/vehicle_parts.json [hunk 3]
```diff
@@ -895,6 +895,20 @@
     },
     {
         "type":"GENERIC",
+        "id" : "hdcamera",
+        "name" : "focused camera",
+        "description" : "A security camera with high quality lenses assembly you could connect to a display.  Allows for greater distance than regular security camera, but the field of vision is limited.",
```
Comment (Cyrano7) [inline data/json/items/vehicle_parts.json]: "A security camera with an assembly of high quality lenses you could connect to a....." Might be a bit better word flow wise but this is just a suggestion.

Comment (tinukedaya) [reply data/json/items/vehicle_parts.json]: That indeed sounds better. I'll change that...

File: data/json/recipes/recipe_electronics.json [hunk 4]
```diff
@@ -2215,6 +2215,39 @@
...
+  "components": [
+    [ [ "omnicamera", 1 ] ],
+    [
+      [ "can_drink_unsealed", 1 ],
+      [ "can_food_unsealed", 1 ],
+      [ "canister_empty", 1 ]
+    ],
+    [ [ "lens", 2 ] ],
+    [
+      [ "superglue", 2 ],
```
[the remark this answers is not in the document]
Comment (tinukedaya) [reply data/json/recipes/recipe_electronics.json]: Oh didn't know about this!

**Comment by tinukedaya** [comment]
Just wondering if I should keep the remark about the limited field of vision, when it's actually not happening in this. Maybe as a flavor text, but it might be confusing I guess....

**Comment by Coolthulhu** [comment]
> Oooh, that seems way out of my league!
> In that case i would just call this done.

I wouldn't.
"Camera+1" crafted by adding scrap to camera is not good enough of an addition.

**Comment by tinukedaya** [comment]
Hmm.. Why not? 
As @OzoneH3 mentioned it is as a TODO in the lightmap, so at some point someone can actually add the calculation and this might be updated with the FoV reduction. 
In the meantime, it's a little buffed camera, only available in one book, with increased battery usage. Having windows on the vehicle is still superior to this and it only opens a little space for tuning. Should I tweak the materials / stats or power consumption to make it acceptable without the FoV? Or just close it straight away for no chance of merging?

**Comment by tinukedaya** [comment]
Update to description and added some more weight to reflect the additional lenses.

**Comment by Coolthulhu** [comment]
> As @OzoneH3 mentioned it is as a TODO in the lightmap, so at some point someone can actually add the calculation and this might be updated with the FoV reduction.

"Maybe someday" is not good enough. Worse: directed items have to be installed as directed items, they can't suddenly become directed after being direction-independent, because they do not have a player-selected direction.

There is no need for a camera+1. If the ordinary camera is too weak, it can be buffed.

**Comment by tinukedaya** [comment]
> Worse: directed items have to be installed as directed items, they can't suddenly become directed after being direction-independent, because they do not have a player-selected direction.

Ouch... Did not consider this...  
Then I will close this one buff the range on the normal security cam slightly to make it a bit more usable.

**Pull request closed without merging**